docs/developer-guide.md

```diff
@@ -0,0 +1,23 @@
+## Dev Workflow
+* Checkout the repo (this assumes the basic GitHub setup already completed)
+  ```
+  git clone https://github.com/GoogleCloudPlatform/anthos-service-mesh-packages.git
+  ```
+
+* Create a new branch (_outlining feature branch workflow as an example here_)
```

Comment:
Actually what I would do is to fork the repo and create the branch there in my own repo? Otherwise the asm repo will be filled with all contributors branches since they have to push. Not sure if this is what we wanted.

Reply:
updated to suggest forking instead